crates/math/src/prev.rs

```diff
@@ -0,0 +1,96 @@
+use {
+    crate::{
+        Dec128, Dec256, Inner, Int128, Int256, Int512, Int64, MathError, MathResult, Udec128,
+        Udec256, Uint128, Uint256, Uint512, Uint64,
+    },
+    bnum::BTryFrom,
+};
+
+/// Describes a number type can be cast into another type of a smaller word size.
+///
+/// For example, [`Uint256`](crate::Uint256) can be cast to [`Uint128`](crate::Uint128).
+/// In this case, [`PrevNumber`] trait should be implemented for [`Uint256`](crate::Uint256)
+/// with `Prev` being [`Uint128`](crate::Uint128).
+pub trait PrevNumber {
+    type Prev;
+    fn into_prev(self) -> MathResult<Self::Prev>;
```

Comment:
1. Add an empty line
2. The method should be called `checked_into_prev`, because it can fail